internal/services/automation/automation_runbook_resource_test.go

```diff
@@ -144,6 +144,31 @@ func TestAccAutomationRunbook_withJobSchedule(t *testing.T) {
 	})
 }
 
+func TestAccAutomationRunbook_withJobScheduleResourceAndUpdate(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_automation_runbook", "test")
+	r := AutomationRunbookResource{}
+	js := AutomationJobScheduleResource{}
+
+	// update Runbook Resource should not cause jobSchedule to be deleted
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.withJobScheduleResource(data, ""),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("publish_content_link"),
+		{
+			Config: r.withJobScheduleResource(data, "Foo"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That("azurerm_automation_job_schedule.test").ExistsInAzure(js),
+			),
+		},
+		data.ImportStep("publish_content_link"),
+	})
+}
+
```

Comment:
Additionally, in the test `TestAccAutomationRunbook_withJobSchedule`, can we test that for a `job_schedule` setting, updating the runbook's other property will works as expected?